agr_literature_service/api/crud/referencefile_crud.py

```diff
@@ -141,6 +141,48 @@ def file_upload(db: Session, metadata: dict, file: UploadFile):  # pragma: no co
     else:
         file_upload_single(db, metadata, file)
 
+    cleanup_temp_file(db, metadata["reference_curie"])
+
+
+def cleanup_temp_file(db: Session, ref_curie: str):  # pragma: no cover
+    if ref_curie is None:
+        raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                            detail="The specified curie is not in the standard Alliance format and no cross "
+                            "references match the specified value.")
+    ref = db.query(ReferenceModel).filter_by(curie=ref_curie).one_or_none()
+    if ref:
+        reffiles = db.query(ReferencefileModel).filter_by(
+            reference_id=ref.reference_id, file_class='main', file_extension='pdf').order_by(
+                ReferencefileModel.file_publication_status).all()
+
+        if len(reffiles) >= 2:
+            modsWithFinal = []
+            pmcFinalPDF = False
+            for x in reffiles:
+                if x.file_publication_status == 'final':
+                    for m in x.referencefile_mods:
+                        if m.mod_id:
+                            if m.mod_id not in modsWithFinal:
+                                modsWithFinal.append(m.mod_id)
+                        else:
+                            pmcFinalPDF = True
+                if x.file_publication_status == 'temp':
+                    toDelete = False
+                    if pmcFinalPDF is True:
+                        toDelete = True
+                    else:
+                        for m in x.referencefile_mods:
+                            if m.mod_id in modsWithFinal:
```

Comment:
This code is marking the file for deletion if there's at least one mod with a final file, but the final file might not be accessible by the other mods. We should delete only when all mods have access to a final pdf, or delete only the connection with the mod (maybe using the `destroy` function in this module?)

Reply:
I'm thinking about something like this:

```reffiles = db.query(ReferencefileModel).filter_by(
            reference_id=ref.reference_id, file_class='main', file_extension='pdf').order_by(
                ReferencefileModel.file_publication_status).all()

        if len(reffiles) >= 2:
            mod_ids_with_final = {mod.mod_id for reffile in reffiles for mod in reffile.referencefile_mods if
                                  reffile.file_publication_status == 'final'}
            for reffile in reffiles:
                if reffile.file_publication_status == 'temp':
                    if None in mod_ids_with_final or all([mod.mod_id in mod_ids_with_final for mod in
                                                          reffile.referencefile_mods]):
                        destroy(db, reffile.referencefile_id)```

Reply:
oh, and the same file (same md5sum) may be shared by other references, so we must use the `destroy` function

Reply:
Wow! it makes sense and the code is so much cleaner. Thanks ! @valearna 
I have implemented it. 

At first, I believed there would be no problem with the temp file uploaded by a mod being replaced by the final file provided by the same mod, but I guess if the paper is a shared one it would be a problem. wait...

Reply:
I need to think some more, but your implement should be safer.

Reply:
"oh, and the same file (same md5sum) may be shared by other references, so we must use the destroy function"

Oh yes. I missed different papers can share files